CMake error when using version 3.10

**Describe the bug**
On older versions of CMake, an OBJECT library cannot link to anything.  However this is used in one spot causing a failure when using v3.10

**To Reproduce**
Attempt to build with cmake v3.10, the following error happens:

```
CMake Error at modules/config/tool/CMakeLists.txt:46 (target_link_libraries):
  Object library target "cfeconfig_platformdata_native_default_cpu1" may not
  link to anything.
```

**Expected behavior**
Should build successfully

**System observed on:**
Ubuntu (old version running in a container)

**Additional context**
We advertise compatibility with 3.10 and up so this should work even though its an obsolete version

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/modules/config/tool/CMakeLists.txt
+++ b/modules/config/tool/CMakeLists.txt
@@ -39,12 +39,11 @@ foreach(SYSVAR ${TGTSYS_LIST})
   )
   target_compile_definitions(${OBJLIB_NAME} PRIVATE
     MAP_ENTRY_LIST=${OBJECT_NAME}
+    $<TARGET_PROPERTY:core_api,INTERFACE_COMPILE_DEFINITIONS>
   )
   target_include_directories(${OBJLIB_NAME} PRIVATE
     ${TEMP_ARCH_BINARY_DIR}/inc
-  )
-  target_link_libraries(${OBJLIB_NAME} PRIVATE
-    core_api
+    $<TARGET_PROPERTY:core_api,INTERFACE_INCLUDE_DIRECTORIES>
   )
 
   # If this is an EDS build, this needs to be built after the EDS tool runs

